### ISSUE TITLE
set_cursor_position [OSX - objc] - not implemented.

### ISSUE THREAD

**Issue by jellytux** [tomaka/glutin#670]
Not implemented. Could someone implement it?

**Comment by fkaa** [comment]
This should be a simple matter of adding a call to `CGWarpMouseCursorPosition`, but some quick googling suggests that [it's a bit trickier than that](http://stackoverflow.com/a/17559012).

**Comment by bjz** [comment]
Here is the glfw implementation: https://github.com/glfw/glfw/blob/e046d0696b6f2760101cebd35ae20e7cb29646b8/src/cocoa_window.m#L1129

```c

void _glfwPlatformSetCursorPos(_GLFWwindow* window, double x, double y)
{
    _glfwPlatformSetCursorMode(window, window->cursorMode);

    const NSRect contentRect = [window->ns.view frame];
    const NSPoint pos = [window->ns.object mouseLocationOutsideOfEventStream];

    window->ns.warpDeltaX += x - pos.x;
    window->ns.warpDeltaY += y - contentRect.size.height + pos.y;

    if (window->monitor)
    {
        CGDisplayMoveCursorToPoint(window->monitor->ns.displayID,
                                   CGPointMake(x, y));
    }
    else
    {
        const NSRect localRect = NSMakeRect(x, contentRect.size.height - y - 1, 0, 0);
        const NSRect globalRect = [window->ns.object convertRectToScreen:localRect];
        const NSPoint globalPoint = globalRect.origin;

        CGWarpMouseCursorPosition(CGPointMake(globalPoint.x,
                                              transformY(globalPoint.y)));
    }
}
```

**Comment by gfetco** [comment]
I ended up just calling CGDisplayMoveCursorToPoint from the core-graphics crate yesterday. I an new to Rust and will happily get back to this after I've gained some more experience on how to communicate with Object-C e.g how to call basic functions, with msg_send!. I'll leave this issue open incase anyone else wants to take a swing at it.

**Comment by fkaa** [comment]
Fixed in #684
